src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2374,6 +2314,8 @@ internal IAsyncResult BeginConnect(EndPoint remoteEP, AsyncCallback callback, ob
                     throw new NotSupportedException(SR.net_invalidversion);
                 }
 
+                ThrowIfNotSupportsMultipleConnectAttempts();
```

Comment:
Isn't this covered by the change below in `BeginConnect(string host, int port, ...)`?